lesson1/ternal_if.js

```diff
@@ -0,0 +1,36 @@
+console.log('Проверка прав с помощью тернального оператора if')
+
+function checkAccess(user) {
+    (user.isAdmin && (user.age >= 18 && user.age <= 35)) ||
+      ((user.age >= 18 && user.age <= 35) && user.paid && !user.blocked && !user.badUsername) ? console.log('Доступ разрешен.') : console.log('Доступ запрещен.')
```

Comment:
Для повышения читаемости кода, при использовании тернарных операторов, можно использовать переменные*:
```suggestion
    const isAgeOk = user.age >= 18 && user.age <= 35;
    const isAdmin = user.isAdmin;
    const paid = user.paid;
    const blocked = user.blocked;
    const badUsername = user.badUsername;
    return isAgeOk && (isAdmin || paid && !blocked && !badUsername) ? console.log('Доступ разрешен.') : console.log('Доступ запрещен.');
```
Вместо блока с переменными еще можно использовать деструкцию объекта**:
```
function hasAccessTern({ age, isAdmin, paid, blocked, badUsername }) {
```

*не обязательно, но лучше делать тернарники максимально информативными
**не обязательно, но тоже повышает читабельность кода

Reply:
Про деструкцию вот не знал, интересный момент

Reply:
Видимо это будет дальше, мы забежали немного наперед =)